src/Sulu/Bundle/WebsiteBundle/Sitemap/SitemapGenerator.php

```diff
@@ -62,7 +62,30 @@ public function generateAllLocals($webspaceKey, $flat = false)
      */
     public function generate($webspaceKey, $locale, $flat = false)
     {
-        $result = $this->contentQuery->execute($webspaceKey, array($locale), $this->contentQueryBuilder, $flat);
+        return $this->executeContentQuery($webspaceKey, array($locale), $flat);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generateForPortal($webspaceKey, $portalKey, $flat = false)
```

Comment:
Is there any use case where it makes sense to generate the sitemap for the entire webspace, instead only for one portal? If yes I would have adapted the existing generate-function, instead of creating a new one.